File: python/src/wdl_model/model/errors/wdl_import_error.py
"""Import-resolution diagnostic emitted by resolver failures."""

from dataclasses import dataclass

from .wdl_error import WdlError


@dataclass
class WdlImportError(WdlError):
    """Diagnostic describing a failure while resolving or loading an imported WDL source."""

    importLocation: str = ""

    def __init__(self, message: str, importLocation: str):
        super().__init__(message=message, line=-1, charPositionInLine=-1)
        self.importLocation = importLocation

    def toDebugMessage(self) -> str:
        return f"{self.__class__.__name__}:{self.importLocation}:{self.message}"

File: typescript/src/model/errors/wdl-import-error.ts
import { WdlError } from './wdl-error.js';

/** Diagnostic describing a failure while resolving or loading an imported WDL source. */
export class WdlImportError extends WdlError {
  public constructor(
    message: string,
    public readonly importLocation: string,
  ) {
    super(message, -1, -1);
  }

  public override toDebugMessage(): string {
    return `${this.constructor.name}:${this.importLocation}:${this.message}`;
  }
}
